### PR TITLE
Script: ulong support via Fields API

### LINKED ISSUE #64361: Support unsigned long field type in painless
The casting model needs to be overhauled to support the `unsigned long` field type, added in [60050](https://github.com/elastic/elasticsearch/pull/60050), as this type can return a `BigInteger`.

Type promotion in the casting model works well for primitive types, String and `def`.

The `def` type is tricky to handle with boxed types and type promotion, the current model has limited support for boxed types.

As this is a major rework on a subtle subsystem, we should look toward interim support for unsigned long.

@jimczi @mayya-sharipova what do you think about having `unsigned long` field return `double`s for `ScriptDocValues`?

Relates: https://github.com/elastic/elasticsearch/issues/64347

### PR DESCRIPTION
**Pull request by stu-elastic**
Exposes unsigned long via the fields API.

Users must choose how to consume unsigned longs, either as regular java `long`s, where
the upper half of the range is treated as negative, or as `BigIntegers`.  

Fields API now requires explicit typing until we do some work in painless work to allow duck
typing.

`script_fields` is not yet implemented with the fields API.

```
field('ul').asBigIntegerField().getValue(BigInteger.valueOf(1000))
field('ul').asBigInteger(BigInteger.valueOf(1000))
field('ul').asLongField().getValue(1000L)
field('ul').asLong(1000L)
```

Fixes: #64361

### CODE REVIEW

File: modules/lang-painless/src/main/resources/org/elasticsearch/painless/spi/org.elasticsearch.script.fields.txt [hunk 1]
```diff
@@ -13,7 +13,41 @@ class org.elasticsearch.script.Field {
...
+
+class org.elasticsearch.script.BigIntegerField {
+  BigInteger getValue(BigInteger)
+}
+
+class org.elasticsearch.script.StringField {
+  String getValue(String)
+}
+
+class org.elasticsearch.script.ObjectField {
```
Comment (jdconrad) [inline modules/lang-painless/src/main/resources/org/elasticsearch/painless/spi/org.elasticsearch.script.fields.txt:49]: Please change this from Object to def to support duck-typing as opposed to just the base Java Object which doesn't make sense to cover most extraneous cases.

**Review by jdconrad — COMMENTED** [review]
This looks really good to me. One minor request - otherwise, I will leave the overall design review to @rjernst.

File: modules/lang-painless/src/main/resources/org/elasticsearch/painless/spi/org.elasticsearch.script.fields.txt (hunk 1 above)
Comment (stu-elastic) [reply modules/lang-painless/src/main/resources/org/elasticsearch/painless/spi/org.elasticsearch.script.fields.txt]: Done.

**Comment by stu-elastic** [comment]
closed in favor of #76519

**Pull request closed without merging**